### PR TITLE
fix #11804

### LINKED ISSUE #11804: Compile-time illegal conversion from int32 to [0..65535] on set `contains`
Passing an int32 out of int16 range seems to work a runtime but it does not at compiletime.

### Example

```nim
proc fooFunc(c: int): bool =
  result = c in {'a'.ord .. 'z'.ord}

static:
  var c = 194708
  echo fooFunc(c)
```

### Current Output
```
Error: illegal conversion from '194708' to '[0..65535]'
```

### Expected Output

```
false
```

### Additional Information

```
Nim Compiler Version 0.20.99 [Linux: amd64]
Compiled at 2019-07-18
Copyright (c) 2006-2019 by Andreas Rumpf

active boot switches: -d:release
```

### PR DESCRIPTION
**Pull request by bung87**
fix #11804

### CODE REVIEW

**Comment by ringabout** [comment]
It seems to work on runtime. It is probably better to "fix" the VM instead of introducing a special case.

```nim
var c = 1
var x: set[char] = {'a'}
echo c in x
```
It also makes this compile, which may not be desired, that seems to break type safety.

**Comment by bung87** [comment]
> It seems to work on runtime. It is probably better to "fix" the VM instead of introducing a special case.
> 
> ```nim
> proc fooFunc(c: int): bool =
>   result = c in {'a'.ord .. 'z'.ord}
> 
> var c = 194708
> echo fooFunc(c)
> ```

I understand that, but I dont have enough knowledge to fix this in vm besides `contains` allowed to overload

File: lib/system/setops.nim [hunk 1]
```diff
@@ -87,3 +87,9 @@ func contains*[T](x: set[T], y: T): bool {.magic: "InSet".} =
   ## `set[char]`! The solution is to bind `T` to `range['a'..'z']`. This
   ## is achieved by reversing the parameters for `contains`; `in` then
   ## passes its arguments in reverse order.
+
+func contains*[T](x: set[T], y: int): bool =
```
Comment (Araq) [inline lib/system/setops.nim:91]: Pretty bad idea, so `4 in {'a'..'z'}` compiles then? What about type safety?

Comment (bung87) [reply lib/system/setops.nim:91]: indeed, close then.

**Pull request closed without merging**